ui/src/api.ts

```diff
@@ -123,3 +117,36 @@ async function doGet<T>(get: Promise<AxiosResponse<T>>) {
   }
   return null;
 }
+
+/** 
+ * helper function for slicing arrays.
+ */
+function sliceArray(array: string[], size: number) {
+  var result = [];
+
+  for (var i = 0; i < array.length; i += size) {
+    result.push(array.slice(i, i+size));
+  }
+
+  return result
+}
+
+/** 
+ * Collect array of promises generated based on array of paper ids.
+ */
+function collectPromise(papers:string[], size:number) {
```

Comment:
Similarly to the comment for `sliceArray`, can you try just putting this into the body of the `papers` method? I think the functionality is small enough and relevant enough to fetch papers that it might make sense to combine it all in one place.